src/Microsoft.AspNetCore.Razor.Language/Legacy/TagHelperParseTreeRewriter.cs

```diff
@@ -817,6 +817,14 @@ private static bool IsEndTag(Block tagBlock)
             return relevantSymbol.Type == HtmlSymbolType.ForwardSlash;
         }
 
+        private static bool IsComment(Span span)
+        {
+            bool isHtmlComment = span.Content?.StartsWith("<!--") == true;
```

Comment:
compare symbols. Using `.Content` is actually quite expensive. Can also simplify the execution path:

```C#
if (IBeHtmlCommentLol)
{
    // HTML comment <!-- ... -->
    return true;
}

if (IBeRazorCommentLol)
{
    // Razor comment @* .. *@
    return true;
}

return false
```

Reply:
Also to note, we use `var` instead of the type names

Reply:
Fixed